lib/rouge/lexers/batch.rb

```diff
@@ -0,0 +1,165 @@
+        cscript curl
+        debug defrag delprof deltree devcon diamond dirquota diruse
+        diskpart diskshadow diskuse dism dnscmd doskey driverquery
+        dsacls dsadd dsget dsquery dsmod dsmove dsrm dsmgmt dsregcmd
+        edlin eventcreate expand extract
+        fc fdisk find findstr fltmc forfiles format freedisk fsutil ftp
+        getmac gpresult gpupdate help hostname
+        icacls ifmember inuse ipconfig kill
+        label lgpo lodctr logman logoff logtime
+        makecab mapisend mbsacli mem
+        mklink mode more mountvol moveuser msg mshta msiexec msinfo32 mstsc
+        nbtstat net net1 netdom netsh netstat nlsinfo nltest now nslookup
+        ntbackup ntdsutil ntoskrnl ntrights nvspbind
+        openfiles
+        pathping perms ping popd portqry powercfg pngout
+        pnputil print printbrm prncnfg prnmngr procdump
+        psexec psfile psgetsid psinfo pskill pslist
+        psloggedon psloglist pspasswd psping
+        psservice psshutdown pssuspend
+        qbasic qgrep qprocess query quser qwinsta
+        rasdial recover reg reg1 regdump regedt32
+        regsvr32 regini replace reset restore
+        rundll32 rmtshare robocopy route rpcping run runas
+        sc scandisk schtasks setspn setx sfc
+        share shellrunas shortcut shutdown
+        sigcheck sleep slmgr sort strings subinacl
+        subst sysmon systeminfo
+        takeown taskkill tasklist telnet tftp
+        timeout tlist touch tracerpt tracert tree tscon
+        tsdiscon tskill tttracer typeperf tzutil
+        undelete unformat
+        verifier vmconnect vssadmin
+        w32tm waitfor wbadmin wecutil wevtutil wget
+        where whoami windiff winrm winrs wmic wpeutil wpr wusa wuauclt
+        wscript
+        xcopy
+      ).join('|')
+
+      state :basic do
+        # Comments
+        rule %r/\b(REM)\b.*$/i, Comment
+        # Empty Labels
+        rule %r/^::.*$/, Comment
+
+        # Labels
+        rule %r/:[a-z]+/i, Keyword::Constant
+
+        # Devices
+        rule %r/@?\b(#{DEVICES})\s*\b/i, Keyword::Reserved
+        rule %r/@?\b(#{DEVICESNUM}[0-9])\s*\b/i, Keyword::Reserved
+
+        # Lang Keywords
+        rule %r/@?\b(#{KEYWORDS})\s*\b/i, Name::Builtin::Pseudo
+        # Internal Commands
+        rule %r/@?\b(#{INTERNAL})\s*\b/i, Name::Builtin::Pseudo
+        # External Commands
+        rule %r/\b(#{EXTERNAL})(\.exe|\.com)?\s*\b/i, Name::Builtin::Pseudo
+
+        # Arguments to commands
+        rule %r/\b(#{ATTRIBUTES})\s*\b/i, Name::Attribute
+        rule %r/([\/\-+][A-Z]+)\s*/i, Name::Attribute
+
+        # Variable Expansions
+        rule %r/[\%!]+([a-z_$@#]+)[\%!]+/i, Name::Variable
+        # For Variables
+        rule %r/(\%+~?[A-Z]+\d?)/i, Keyword::Constant
+
+        rule %r/\b(set)\b/i, Keyword::Declaration
+
+        # Operators
+        rule %r/[%!()=<>&|~$*@]/, Operator
+
+      end
+
+      state :single_quotes do
+        rule %r/[']/, Str::Single, :pop!
```

Comment:
There's no way to escape quotes?

Reply:
In the for /f is possible use quotes inside a double quote text:
`for /f %%a in ("-"hi"-") do echo %%~a`
produces:
`-"hi"-`
Is not a common syntax.